app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java

```diff
@@ -22,28 +23,84 @@
 import com.google.android.gms.maps.model.MapStyleOptions;
 import com.google.android.gms.maps.model.Marker;
 import com.google.android.gms.maps.model.MarkerOptions;
+
 import java.util.ArrayList;
 
 public class NavigationFragment extends Fragment implements OnMapReadyCallback, OnCameraIdleListener {
 
+    private enum ViewType {
+        INDOOR, OUTDOOR
+    }
+
     private static final MarkerOptions LOYOLA_MARKER = new MarkerOptions().position(Campus.LOYOLA.getMapCoordinates()).title(Campus.LOYOLA.getName());
     private static final MarkerOptions SGW_MARKER = new MarkerOptions().position(Campus.SGW.getMapCoordinates()).title(Campus.SGW.getName());
     private static final float streetLevelZoom = 15f;
-    Campus currentCampus = Campus.SGW;
-    GoogleMap map;
+    private GoogleMap map;
+
+    //State
+    private ViewType viewType;
+    private Campus currentCampus = Campus.SGW;
     private ArrayList<Marker> buildingMarkers = new ArrayList<>();
```

Comment:
List<String> arrayList = new ArrayList<String>();
Is generic where you want to hide implementation details while returning it to client, at later point of time you may change implementation from ArrayList to LinkedList transparently.

This mechanism is useful in cases where you design libraries etc., which may change their implementation details at some point of time with minimal changes on client side.

ArrayList<String> arrayList = new ArrayList<String>();
This mandates you always need to return ArrayList. At some point of time if you would like to change implementation details to LinkedList, there should be changes on client side also to use LinkedList instead of ArrayList.